### PR TITLE
Issue 428 v1 sending rows and cols

### PR DESCRIPTION
**Pull request by DmitryPorotov**
Send number of rows and columns so the viewer sees the same size window.

### CODE REVIEW

File: src/gate.ts [hunk 1]
```diff
@@ -391,7 +391,7 @@ export class Gate {
     setLayout(state: ServerPayload = null) {
         console.log("in setLayout", state)
         const winLen = this.windows.length
-        this.fontScale = 1
+        if(this.fitScreen) this.fontScale = 1
```
Comment (daonb) [inline src/gate.ts:394]: please split to two lines

File: src/gate.ts [hunk 2]
```diff
@@ -476,7 +476,9 @@ export class Gate {
             container.style.transformOrigin = "top left"
         }
         this.panes().forEach(p => {
-            p.t.options.fontSize = Math.floor(scale * p.fontSize)
+            const hasFraction = String(p.fontSize * scale).includes('.')
```
Comment (daonb) [inline src/gate.ts:479]: Please add a comment explaining the three lines

File: src/pane.ts [hunk 3]
```diff
@@ -298,6 +312,8 @@ export class Pane extends Cell {
     // returns true is size was changed
     // TODO: make it async
     fit(cb = null) {
+        if (!this.gate?.fitScreen) return;
+        console.log('in pane fit')
```
Comment (daonb) [inline src/pane.ts:316]: Please remove

File: src/window.ts [hunk 4]
```diff
@@ -262,8 +262,25 @@ export class Window {
...
-                thisCell.fit()
+                const hasFraction = String(thatCell.fontSize * this.gate.fontScale).includes('.')
+                thisCell.t.options.fontSize = Math.floor(thatCell.fontSize * this.gate.fontScale) + (hasFraction ? .5 : 0)
+                if (thisCell.fitAddon.proposeDimensions) {
+                    const dims = thisCell.fitAddon.proposeDimensions()
+                    if (dims && (dims.cols !== thatCell.cols || dims.rows !== thatCell.rows)) {
+                        thisCell.t.options.fontSize -= .5
+                    }
+                }
+                thisCell.t._core._renderService.clear()
```
Comment (daonb) [inline src/window.ts:273]: Why clear? It's a bad idea to use `_core` as it's internal and we have no guarantee and it'll work. It also doesn't work on safari.

File: src/window.ts [hunk 5]
```diff
@@ -262,8 +262,25 @@ export class Window {
...
+                thisCell.t.resize(thatCell.cols, thatCell.rows)
+                setTimeout(() => {
+                    if (thisCell instanceof Pane) {
```
Comment (daonb) [inline src/window.ts:276]: Please add a comment explaining this dark magic

File: src/window.ts [hunk 6]
```diff
@@ -275,6 +292,7 @@ export class Window {
             setTimeout(() => zoomed.zoom(), 100)
             console.log("will zoom in 100ms")
         }
+        newLayout.refreshDividers()
```
Comment (daonb) [inline src/window.ts:295]: nice catch

**Review by daonb — CHANGES_REQUESTED** [review]

File: src/window.ts (hunk 5 above)
Comment (DmitryPorotov) [reply src/window.ts:276]: I'll rewrite this without the dark magic if I'm already using the _core anyways.

**Review by daonb — APPROVED** [review]

**Pull request merged**